### ISSUE TITLE
How to use `openCursor` with `autoAdvance: true` for very large datasets?

### ISSUE THREAD

**Issue by ricardoboss** [tekartik/idb_sqflite#11]
Hi there!

I am using this dependency in my app to support both web and native using an indexeddb-like api:

```
  tekartik_app_flutter_idb:
    git:
      url: https://github.com/tekartik/app_flutter_utils.dart
      ref: dart2_3
      path: app_idb
```

This dependency uses this library on iOS, so I figured I had to open an issue here.

---

I ran into a problem with sqflite and very large datasets (40k+ entries). I want to implement a search function and read the object store using `openCursor(autoAdvance: true)`, which gives me a stream of cursors with values which I can then filter.

The problem arises when the full dataset is queried and autoadvance reaches the stack limit, because then it throws a StackOverflow exception, which I cannot handle in the app code, unfortunately.

The search kinda works with small queries (3-4 characters), but completely fails with large ones. And I always get a StackOverflow exception:

<details>
<summary>Output</summary>

```
[VERBOSE-2:ui_dart_state.cc(198)] Unhandled Exception: Stack Overflow
#0      _StreamController._add (dart:async/stream_controller.dart:646:3)
#1      _IdbCursorBaseControllerSqflite.advance (package:idb_sqflite/src/sqflite_cursor.dart:192:13)
#2      _IdbCursorBaseControllerSqflite._autoNext (package:idb_sqflite/src/sqflite_cursor.dart:200:9)
#3      _IdbCursorBaseControllerSqflite._autoNext (package:idb_sqflite/src/sqflite_cursor.dart:202:9)
#4      _IdbCursorBaseControllerSqflite._autoNext (package:idb_sqflite/src/sqflite_cursor.dart:202:9)
#5      _IdbCursorBaseControllerSqflite._autoNext (package:idb_sqflite/src/sqflite_cursor.dart:202:9)
#6      _IdbCursorBaseControllerSqflite._autoNext (package:idb_sqflite/src/sqflite_cursor.dart:202:9)
#7      _IdbCursorBaseControllerSqflite._autoNext (package:idb_sqflite/src/sqflite_cursor.dart:202:9)
#8      _IdbCursorBaseControllerSqflite._autoNext (package:idb_sqflite/src/sqflite_cursor.dart:202:9)
#9      _IdbCursorBaseControllerSqflite._autoNext (packag<…>

```

</details>

I identified this block of code to be the culprit: https://github.com/tekartik/idb_sqflite/blob/dart2_3/idb_sqflite/lib/src/sqflite_cursor.dart#L199

Any idea what I can do?

**Comment by alextekartik** [comment]
Thanks for the report and the detailed (correct) analysis. It has to be fixed on my side, I don't think that we have to be synchronous here so I'll make some tests.
I'll try to reproduce first in another environment than iOS for easier analysis

**Comment by alextekartik** [comment]
Should be fixed in idb_sqlite 1.1.1 (pub upgrade should get it)

**Comment by ricardoboss** [comment]
I was able to reproduce the problem on an Android emulator and after a `pub upgrade`, it works like a charm :)

Thank you very much for the quick fix!
